src/Congo.WebApi/Data/CartAccess/AddToCartHandler.cs

```diff
@@ -0,0 +1,50 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Congo.WebApi.Data.Models;
+using MediatR;
+using Microsoft.EntityFrameworkCore;
+using static Congo.WebApi.Data.CartAccess.CartCommands;
+
+namespace Congo.WebApi.Data.CartAccess
+{
+    public class AddToCartHandler : IRequestHandler<AddToCartCommand, Guid>
+    {
+        private readonly CongoContext _dbContext;
+        private Cart _cart;
+
+        public AddToCartHandler(CongoContext dbContext)
+        {
+            _dbContext = dbContext;
+        }
+
+        public async Task<Guid> Handle(AddToCartCommand request, CancellationToken cancellationToken)
+        {
+            
+            if (request.cartId != null)
+            {
+                _cart = await _dbContext.Carts.FirstOrDefaultAsync(c => c.Id == request.cartId);
+            }
+            else
+            {
+                _cart = new Cart();
+                await _dbContext.Carts.AddAsync(_cart);
+            }
+                
+            var cartItem = new CartItem
+            {
+                Id = request.product.Id,
+                CartId = _cart.Id,
```

Comment:
Cart may be null here if we pass in an invalid `cartId`, you will want to check for this

Reply:
the check was above, where if there was no cartId passed in, a new cartId will be created```
 public async Task<Guid> Handle(AddToCartCommand request, CancellationToken cancellationToken)
        {
            
            if (request.cartId != null)
            {
                _cart = await _dbContext.Carts.FirstOrDefaultAsync(c => c.Id == request.cartId);
            }
            else
            {
                _cart = new Cart(){Id = Guid.NewGuid()};
                await _dbContext.Carts.AddAsync(_cart);
            }
```

Reply:
Yup that's right, `if there was no cartId passed in, a new cartId will be created` - this part is correct. But what happens if I pass in a cart id that doesn't exist here?

Reply:
Would it be better to return no content in this case or return a new Cart?

Reply:
I think we'd want to error here? Then have the api return that error.
Not entirely sure where we want to do validation stuff.
@pobiega want to chime in?

Reply:
I prefer the error route, let the frontend handle it (they sent the bad value, they get to try and solve it).